File: lib/repositories/berth-repository.ts
import { readFileSync } from 'fs'
import { join } from 'path'

interface BerthLocation {
  lat: number
  lng: number
}

export class BerthRepository {
  private readonly berths: Map<string, BerthLocation>

  constructor() {
    const path = join(process.cwd(), 'data', 'berths.json')
    const raw = JSON.parse(readFileSync(path, 'utf-8')) as Record<string, BerthLocation>
    this.berths = new Map(Object.entries(raw).map(([k, v]) => [k.toLowerCase(), v]))
  }

  get(areaId: string, berthId: string): BerthLocation | null {
    return this.berths.get(`${areaId}_${berthId}`.toLowerCase()) ?? null
  }
}

File: api/Services/BerthRepository.cs
using System.Text.Json;

namespace Api.Services;

public record BerthLocation(double Lat, double Lng);

public class BerthRepository
{
    // Key: "AREAID_BERTHID" e.g. "WX_0100"
    private readonly Dictionary<string, BerthLocation> _berths;

    public BerthRepository(IWebHostEnvironment env)
    {
        var path = Path.Combine(env.ContentRootPath, "Data", "berths.json");
        if (File.Exists(path))
        {
            var json = File.ReadAllText(path);
            _berths = JsonSerializer.Deserialize<Dictionary<string, BerthLocation>>(json)
                      ?? new Dictionary<string, BerthLocation>();
        }
        else
        {
            _berths = new Dictionary<string, BerthLocation>();
        }
    }

    public BerthLocation? Get(string areaId, string berthId) =>
        _berths.TryGetValue($"{areaId}_{berthId}", out var loc) ? loc : null;
}
